adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -96,4 +112,38 @@ protected String getTypeName() {
         }
         return result.future();
     }
+
+    private Future<ResourceIdentifier> mapMessage(final MqttContext ctx, final ResourceIdentifier targetAddress, final MqttPublishMessage message) {
+        return getRegistrationAssertion(targetAddress.getTenantId(), targetAddress.getResourceId(),
+                ctx.authenticatedDevice(), ctx.getTracingContext()).compose(registratonInfo -> {
+            final Promise<ResourceIdentifier> result = Promise.promise();
+            if (!registratonInfo.containsKey(RegistrationConstants.FIELD_MAPPER)) {
+                result.complete(targetAddress);
+                return result.future();
+            }
+            final String mapper = registratonInfo.getString(RegistrationConstants.FIELD_MAPPER);
+            if (mapper.isBlank()) {
+                result.complete(targetAddress);
+                return result.future();
+            }
+            final HashMap<String, MapperEndpoint> mapperEndpoints = getConfig().getMapperEndpoints();
+            if (!mapperEndpoints.containsKey(mapper)) {
+                result.complete(targetAddress);
+                return result.future();
+            }
+            final MapperEndpoint mapperEndpoint = mapperEndpoints.get(mapper);
+            final WebClient webClient = WebClient.create(vertx);
+            webClient.get(mapperEndpoint.getPort(), mapperEndpoint.getHost(), mapperEndpoint.getUri())
+                    .addQueryParam("payload", getHex(message.payload().getBytes()))
```

Comment:
I would rather do a POST and include the payload as a plain byte array in the request body so that there is no need to encode the data to a HEX string.

Reply:
I agree, better this way